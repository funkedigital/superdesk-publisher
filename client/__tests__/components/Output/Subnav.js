import React from "react";
import Subnav from "../../../components/Output/Subnav";
import { render, fireEvent } from "@testing-library/react";
import Store from "../../../components/Output/Store";

const languages = [
  { qcode: 'pl', name: 'Polish' },
  { qcode: 'en', name: 'English' },
  { qcode: 'de', name: 'German' },
]

describe("Output/Subnav", () => {

  it("renders correctly", () => {
    const { container } = render(
      <Store.Provider
        value={{
          articlesCounts: { incoming: 0, published: 0 },
          tenants: [
            { name: 'tenant1', code: 'tenant1' },
            { name: 'tenant2', code: 'tenant2' }
          ],
          filters: {},
          selectedList: 'incoming',
          actions: { setSelectedList: jest.fn() }
        }}
      >
        <Subnav isLanguagesEnabled={true}
          languages={languages} />
      </Store.Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("sets selected list", async () => {
    const setList = jest.fn();

    const { getByText } = render(
      <Store.Provider
        value={{
          articlesCounts: { incoming: 0, published: 0 },
          tenants: [
            { name: 'tenant1', code: 'tenant1' },
            { name: 'tenant2', code: 'tenant2' }
          ],
          filters: {},
          selectedList: 'incoming',
          actions: { setSelectedList: setList }
        }}
      >
        <Subnav isLanguagesEnabled={true}
          languages={languages} />
      </Store.Provider>
    );

    const button = getByText('Published');
    fireEvent.click(button);

    expect(setList).toHaveBeenCalled();
    expect(setList).toHaveBeenCalledWith("published");
  });
});
