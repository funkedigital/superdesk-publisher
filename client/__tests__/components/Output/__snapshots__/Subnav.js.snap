// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Output/Subnav renders correctly 1`] = `
<div
  class="subnav subnav--lower-z-index"
>
  <div
    class="subnav__content-bar"
  >
    <span
      class="sd-check__wrapper"
    >
      <span
        class="sd-checkbox sd-checkbox--button-style sd-checkbox--radio checked"
      >
        <label>
          Incoming content
        </label>
      </span>
    </span>
    <span
      class="sd-check__wrapper"
    >
      <span
        class="sd-checkbox sd-checkbox--button-style sd-checkbox--radio"
      >
        <label>
          Published
        </label>
      </span>
    </span>
  </div>
  <div
    class="subnav__content-bar sd-flex-wrap ml-auto sd-padding-l--1"
  >
    <div
      class="dropdown dropdown--align-right"
      flow="left"
    >
      <span>
        <button
          class="dropdown__toggle"
        >
          Created
          <span
            class="dropdown__caret"
          />
        </button>
      </span>
      <ul
        class="dropdown__menu dropdown--align-right"
      >
        <li>
          <button>
            Updated
          </button>
        </li>
        <li>
          <button>
            Created
          </button>
        </li>
      </ul>
    </div>
    <a
      class="icn-btn"
      flow="bottom"
      sd-tooltip="Ascending"
    >
      <i
        class="icon-ascending"
      />
    </a>
  </div>
  <div
    class="subnav__spacer subnav__spacer--no-margin"
  />
  <div
    class="subnav__content-bar sd-flex-wrap sd-padding-l--1 ui-responsive-small--d-none ui-responsive-small--d-block if-authoring--d-none"
  >
    <div
      class="sd-margin-r--1"
    >
      <span>
        Incoming:
      </span>
      <span
        class="badge sd-margin-l--0-5"
      >
        0
      </span>
    </div>
    <div>
      <span>
        Published:
      </span>
      <span
        class="badge sd-margin-l--0-5"
      >
        0
      </span>
    </div>
  </div>
</div>
`;
